src/lib/dialog/dialog.ts

```diff
@@ -120,8 +120,8 @@ export class MatDialog {
    * @param config Extra configuration options.
    * @returns Reference to the newly-opened dialog.
    */
-  open<T, D = any, R = any>(componentOrTemplateRef: ComponentType<T> | TemplateRef<T>,
-          config?: MatDialogConfig<D>): MatDialogRef<T, R> {
+  open<T, D = any>(componentOrTemplateRef: ComponentType<T> | TemplateRef<T>,
+                   config?: MatDialogConfig<D>): MatDialogRef<T> {
```

Comment:
Note: the original issue called for the return type to be changed to `MatDialogRef<T, {}>`, but I'm not sure that it makes sense. It means that people wouldn't be able to do something like `dialogRef.afterClosed().subscribe((result: number) => ...)`. I was getting type errors when trying to cast explicitly, as well (e.g. `dialog.open(...) as MatDialogRef<SomeComponent, number>`).